### PR TITLE
fix bug in ocn2glc_coupling and add blom 

### PR DESCRIPTION
**Pull request by mvertens**
### Description of changes
This adds blom as another ocean component in cime_config.

### Specific notes
This adds blom as another ocean component in cime_config and also fixes the query of the logical variable for ocn2glc coupling in esmFldsExchange_cesm.F90.

Contributors other than yourself, if any: None

CMEPS Issues Fixed: No

Are changes expected to change answers? bfb

Any User Interface Changes (namelist or namelist defaults changes)? No

### Testing performed
Verified that ocn2glc coupling is working correctly with blom.

### CODE REVIEW

File: cime_config/config_component_cesm.xml [hunk 1]
```diff
@@ -190,21 +226,23 @@
       <!-- =================================================== -->
       <value compset="_DATM.*_DICE.*_POP2">24</value>
       <value compset="_DATM.*_DICE.*_MOM6">24</value>
-      <!-- TODO: currently a NUOPC runseqence cannot be generated with the ATM_NCPL < OCN_NCPL - as is the case -->
-      <!-- with the POP2 C compset run sequence - for now will set these to be the same as MOM6 -->
+      <value compset="_DATM.*_DICE.*_BLOM">24</value>
+      <!-- NOTE: urrently a NUOPC runseqence cannot be generated with the ATM_NCPL < OCN_NCPL -->
```
Comment (billsacks) [inline cime_config/config_component_cesm.xml:230]: Minor typo: urrently -> currently

File: cime_config/config_component_cesm.xml [hunk 2]
```diff
@@ -141,6 +141,42 @@
...
+
+  <entry id="NH3_EMIS_OCN">
+    <type>logical</type>
+    <default_value>FALSE</default_value>
+    <valid_values>TRUE,FALSE</valid_values>
+    <group>run_coupling</group>
+    <file>env_run.xml</file>
+    <desc>Activates NH3 fluxes to be sent from ocn to atm </desc>
+  </entry>
+
```
Comment (billsacks) [inline cime_config/config_component_cesm.xml:179]: Are these only valid for BLOM currently? If so, I feel this should at least be documented. Ideally they would only be available for BLOM, or there would be an error check to ensure they aren't set for other models, but I suspect that's hard... if it's true that those would be hard to do, then I think it's fine to just add in the description something like "Only available with the BLOM ocean model" or something like that.

**Review by billsacks — CHANGES_REQUESTED** [review]
A couple of questions / minor (I hope) requests below.

**Comment by mvertens** [comment]
@billsacks - thanks so much for your review. I've addressed the issues you raised.

**Review by billsacks — APPROVED** [review]
Looks good - thank you!

**Pull request merged**